### PR TITLE
Add arg builder to legal hold APIs

### PR DESCRIPTION
**Pull request by sinhaashish**

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -3927,15 +3927,52 @@ public void enableObjectLegalHold(String bucketName, String objectName, String v
...
+  public void enableObjectLegalHold(EnableObjectLegalHoldArgs args)
+      throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
+          InternalException, InvalidBucketNameException, InvalidKeyException,
+          InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
     Map<String, String> queryParamMap = new HashMap<>();
     queryParamMap.put("legal-hold", "");
 
-    if (versionId != null && !versionId.isEmpty()) {
-      queryParamMap.put("versionId", versionId);
+    if (args.versionId() != null) {
```
Comment (anjalshireesh) [inline api/src/main/java/io/minio/MinioClient.java:3970]: Was the condition `&& !versionId.isEmpty()` removed intentionally? Seems like a useful validation.
Might be a good idea to add it as a validation (must be non-empty if non-null) in the args builder.

Comment (balamurugana) [reply api/src/main/java/io/minio/MinioClient.java:3970]: As AWS works with empty string as `versionId`, the check is OK

File: examples/DisableObjectLegalHold.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+              "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG");
+
+      // Disable object legal hold.
+      minioClient.disableObjectLegalHold(
+          DisableObjectLegalHoldArgs.builder()
+              .bucket("my-bucketname")
+              .object("my-objectname")
+              .build());
+
+      // Check if the object legal hold is enabled or not.
```
Comment (balamurugana) [inline examples/DisableObjectLegalHold.java:44]: It supposes to be `disabled or not`?

File: examples/DisableObjectLegalHold.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      System.out.println(
+          " Is object legal hold is enabled "
+              + minioClient.isObjectLegalHoldEnabled(
```
Comment (balamurugana) [inline examples/DisableObjectLegalHold.java:47]: We supposes to error out if its not disabled?

File: examples/EnableObjectLegalHold.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+      // Enable object legal hold.
+      minioClient.enableObjectLegalHold(
+          EnableObjectLegalHoldArgs.builder()
+              .bucket("my-bucketname")
+              .object("my-objectname")
+              .versionId("object-versionId")
+              .build());
+
+      // Check if the object legal hold is enabled or not.
```
Comment (balamurugana) [inline examples/EnableObjectLegalHold.java:45]: same as comments in examples/DisableObjectLegalHold.java

File: functional/FunctionalTest.java [hunk 5]
```diff
@@ -3588,15 +3590,15 @@ public static void enableObjectLegalHold_test() throws Exception {
...
-        mintSuccessLog(
-            "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
-            null,
-            startTime);
+        client.disableObjectLegalHold(
+            DisableObjectLegalHoldArgs.builder().bucket(bucketName).object(objectName).build());
+        mintSuccessLog("enableObjectLegalHold(EnableObjectLegalHoldArgs args)", null, startTime);
       } finally {
         client.removeObject(
             RemoveObjectArgs.builder().bucket(bucketName).object(objectName).build());
```
Comment (balamurugana) [inline functional/FunctionalTest.java:3604]: You would need to use `handleException()` in all modifed tests.

File: api/src/main/java/io/minio/MinioClient.java [hunk 6]
```diff
@@ -4211,18 +4289,66 @@ public void disableObjectLegalHold(String bucketName, String objectName, String
    * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
    * @throws XmlParserException thrown to indicate XML parsing error.
    */
+  @Deprecated
```
Comment (anjalshireesh) [inline api/src/main/java/io/minio/MinioClient.java:4292]: It'll be good to add the deprecation javadoc comment also along with a link to the new method to be used

**Pull request merged**